### ISSUE TITLE
Question: Is it possible to implement the transfer of the room name assigned to a device in the CCU to Homekit ?

### ISSUE THREAD

**Issue by jhk3** [thkl/homebridge-homematic#246]
I have noticed that names of devices newly added to the CCU2 (raspberrymatic in my case) are only transferred to the Homekit "Home" App (and the like), if in the CCU2, the name of a certain channel (could be any channel, but I do not remember) of the device is given the same name as the device itself.
 
What is apparently not transferred is the name of the room to which the device is assigned in the CCU. Is it possible to include such a feature? This would be nice to have, as it would eliminate the necessity to do this manually in the Home App (if a room with an identical name already exists in "Home", the device should automatically be added to that room). Also, all device names and rooms would then be immediately correctly assigned in "Home" when a bridge is added in the "Home"-App.

**Comment by thkl** [comment]
Sorry this is not possible. Homekit has (currently) no Service or Characteristic for a room assignment.
Please keep in mind, normally each device is a stand alone device which is talking to HomeKit. This device (eg a Lightbulb does not know anything about its location in your house). The homematic Plugin is a special device (bridge) which contains other devices (ccu channels) behind itself ....

**Comment by jhk3** [comment]
Of course a device does not know where it will be installed. My intention was to assign the rooms in the CCU and then, upon adding the homebridge to homekit, transfer this information from the ccu to homekit. Thus one would have to enter room assignments only once into the CCU (from where it will be taken to homekit. My knowledge about the APIs and how homebridge talks to homekit is too limited, but it would appear to me that the last paragraphs of [https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/HomeKitDeveloperGuide/WritingtotheHomeKitDatabase/WritingtotheHomeKitDatabase.html](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/HomeKitDeveloperGuide/WritingtotheHomeKitDatabase/WritingtotheHomeKitDatabase.html) contain information about adding zones and rooms in homekit. Then again, I am no expert and I also do not know whether such interfaces are supported by the homebridge project.

**Comment by thkl** [comment]
So far homebridge (and as i know the HomeKit Accessory Protocol also) does not support an assignment of rooms or zones for a device. The documentation above is about the iOS side of HomeKit (if you want to build your own Home.app)

**Comment by jhk3** [comment]
Then we rely on Apple that they allow such an automatism during bridge/device import into iOS "Home" App in a future iOS. I briefly checked and it seems that currently (iOS 11.3), new devices are auto-added to the room to which one has relocated the bridge (Homebridge), which in my case is a room with name "Büro".

Happy Easter holidays!

**Comment by RealDeemer** [comment]
Just want to leave here, that the Eve App on the iPhone does support to add rooms and/or assign devices to a room. While that might not be part of the HomeKit protocol, there might be at least an API for that.
